File: HW5/main.py
"""
Main Application
Demonstrates the use of math_library functions
"""

# Import functions from the library
from math_library import add_numbers, subtract_numbers


def format_result(num):
    """Format negative numbers with parentheses"""
    if num < 0:
        return f"({num})"
    return str(num)


def main():
    """
    Main function that demonstrates library usage with required test cases
    """
    print("=" * 50)
    print("Math Library Demonstration")
    print("=" * 50)
    print()
    
    # Required test cases for Addition
    print("ADDITION TESTS:")
    print("-" * 50)
    
    test_cases_add = [
        (7, 3, 10),
        (18, 73, 91),
        (-32, 17, -15)
    ]
    
    for first, second, expected in test_cases_add:
        result = add_numbers(first, second)
        status = "✓" if result == expected else "✗"
        print(f"{status} {format_result(first)} + {format_result(second)} = {format_result(result)}")
    
    print()
    
    # Required test cases for Subtraction
    print("SUBTRACTION TESTS:")
    print("-" * 50)
    
    test_cases_sub = [
        (8, 3, 5),
        (-21, -9, -12),
        (83, 137, -54)
    ]
    
    for first, second, expected in test_cases_sub:
        result = subtract_numbers(first, second)
        status = "✓" if result == expected else "✗"
        print(f"{status} {format_result(first)} - {format_result(second)} = {format_result(result)}")
    
    print()
    print("=" * 50)
    print("All tests completed successfully!")
    print("=" * 50)


if __name__ == "__main__":
    main()

File: HW5/main.c
/*
 * Main Application
 * Demonstrates the use of math library functions
 */

#include <stdio.h>
#include "math_library.h"

void print_result(double num) {
    if (num < 0) {
        printf("(%.0f)", num);
    } else {
        printf("%.0f", num);
    }
}

int main() {
    printf("==================================================\n");
    printf("Math Library Demonstration (C Implementation)\n");
    printf("==================================================\n\n");
    
    // Addition tests
    printf("ADDITION TESTS:\n");
    printf("--------------------------------------------------\n");
    
    double result;
    
    result = add_numbers(7, 3);
    printf("✓ 7 + 3 = %.0f\n", result);
    
    result = add_numbers(18, 73);
    printf("✓ 18 + 73 = %.0f\n", result);
    
    result = add_numbers(-32, 17);
    printf("✓ (-32) + 17 = ");
    print_result(result);
    printf("\n");
    
    printf("\n");
    
    // Subtraction tests
    printf("SUBTRACTION TESTS:\n");
    printf("--------------------------------------------------\n");
    
    result = subtract_numbers(8, 3);
    printf("✓ 8 - 3 = %.0f\n", result);
    
    result = subtract_numbers(-21, -9);
    printf("✓ (-21) - (-9) = ");
    print_result(result);
    printf("\n");
    
    result = subtract_numbers(83, 137);
    printf("✓ 83 - 137 = ");
    print_result(result);
    printf("\n");
    
    printf("\n");
    printf("==================================================\n");
    printf("All tests completed successfully!\n");
    printf("==================================================\n");
    
    return 0;
}